scripts/sign-and-notarize.sh

```diff
@@ -0,0 +1,219 @@
+#!/bin/bash
+
+set -eo pipefail
+
+# This script signs and notarizes release binaries as follows:
+# * unpack the given tarball
+# * create a keychain and import the signing cert
+# * sign each file in the ${RELEASE_DIR}/bin/ directory
+# * zip up the whole provided directory (.tar.gz is not supported by Apple)
+# * submit the zip to Apple for notarizing
+# * verify that all files were notarized
+# * create the release tarball
+
+if [[ $# -ne 2 ]]; then
+  echo "Usage: $0 <release dir> <release tarball>" >&2
+  exit 1
+fi
+
+# Check for required env vars
+if [ -z "${THELMA_MACOS_APP_PWD}" ]; then
+	echo "ERROR: Apple Developer application password env var THELMA_MACOS_APP_PWD unset but required. Exiting."
+	exit 1
+fi
+if [ -z "${THELMA_MACOS_CERT}" ]; then
+	echo "ERROR: Signing cert env var THELMA_MACOS_CERT unset but required. Exiting."
+	exit 1
+fi
+if [ -z "${THELMA_MACOS_CERT_PWD}" ]; then
+	echo "ERROR: Signing cert password env var THELMA_MACOS_CERT unset but required. Exiting."
+	exit 1
+fi
+
+readlinkf(){
+	perl -MCwd -e 'print Cwd::abs_path shift' "$1"
+}
+
+# Files and dirs
+TEMP_DIR=${1}
+RELEASE_TARBALL=${2}
+WORKING_DIR=$(readlinkf "${TEMP_DIR}")/san
+
+# XCode signing info - doesn't contain secrets
+APPLE_ID=appledev@broadinstitute.org
+TEAM_ID=R787A9V6VV
+SECURITY_ID=5784A30A5BFD511E8636B9F6BBE7EE36D0F0A726
+CMD_AUTH_FLAGS="--apple-id ${APPLE_ID} --password ${THELMA_MACOS_APP_PWD} --team-id ${TEAM_ID}"
+
+# Sign one file
+sign() {
+	codesign -f -o runtime,library --timestamp -s "${SECURITY_ID}" "${1}"
+}
+
+# Zip the given directory into the working dir
+archive() {
+	# Get the absolute path to the input path
+	local _absdir="$(readlinkf "${1}")"
+
+	# Extract just the name of the directory to zip
+	local _bname="$(basename ${_absdir})"
+
+	# Save the output filepath
+	local _outfile="${WORKING_DIR}/${_bname}".zip
+
+	# Zip will update in-place so make sure to delete 
+	rm "${_outfile}" 2>/dev/null
+
+	# Go to the parent directory of the input path to avoid directories inside the zip
+	cd "${_absdir}"
+
+	# Create the archive
+	zip -rDq "${_outfile}" *
+
+	# Go back to the previous directory
+	cd - > /dev/null
+
+	# Other fns get the signed zip file from stdout
+	echo "${_outfile}"
+}
+
+# Upload the zip file for notarization and wait for a response
+notarize() {
```

Comment:
Apple notarizes all the files that we upload in the zip, and since notarization requires uploading a file to Apple and waiting for their approval, I think it's best to do it all at once. TL;DR: No, we do not need to notarize each file individually.